File: 09.Greed贪心/leetcode_1464_maxProduct.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int maxProduct(vector<int>& nums) {
        sort(nums.begin(), nums.end(), greater<int>());
        return (nums[0]-1)*(nums[1]-1);
    }
};

int main(){
    Solution s;
    vector<int> nums = {3,4,5,2};
    cout << s.maxProduct(nums) << endl;
}

File: 09.Greed贪心/leetcode_1464_maxProduct.py
from typing import List

class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        nums.sort()
        return (nums[-1]-1)*(nums[-2]-1)

if __name__ == "__main__":
    s = Solution()
    nums = [3,4,5,2]
    print(s.maxProduct(nums))
